db/views/reporting_overdue_patients_v01.sql

```diff
@@ -0,0 +1,194 @@
+WITH patients_with_appointments AS (
+    SELECT
+        DISTINCT ON (rps.patient_id, rps.month_date)
+        rps.month_date,
+        rps.patient_id,
+        rps.hypertension AS hypertension,
+        rps.diabetes AS diabetes,
+        rps.htn_care_state,
+        rps.month,
+        rps.quarter,
+        rps.year,
+        rps.month_string,
+        rps.quarter_string,
+        rps.assigned_facility_id,
+        rps.assigned_facility_slug,
+        rps.assigned_facility_region_id,
+        rps.assigned_block_slug,
+        rps.assigned_block_region_id,
+        rps.assigned_district_slug,
+        rps.assigned_district_region_id,
+        rps.assigned_state_slug,
+        rps.assigned_state_region_id,
+        rps.assigned_organization_slug,
+        rps.assigned_organization_region_id,
+        appointments.id AS previous_appointment_id,
+        appointments.device_created_at AS previous_appointment_date,
+        appointments.scheduled_date AS previous_appointment_schedule_date
+    FROM reporting_patient_states rps
+    LEFT JOIN appointments ON appointments.patient_id = rps.patient_id
+        AND appointments.device_created_at < rps.month_date
+    WHERE rps.status <> 'dead'
+	AND rps.month_date > NOW() - INTERVAL '24 months'
+    ORDER BY
+        rps.patient_id,
+        rps.month_date,
+        appointments.device_created_at DESC
+),
+
+patients_with_appointments_and_visits AS (
+    SELECT
+        DISTINCT ON ( patients_with_appointments.patient_id, patients_with_appointments.month_date)
+        patients_with_appointments.*,
+        visit_id,
+        visited_at_after_appointment
+    FROM patients_with_appointments
+    LEFT JOIN lateral (
+        SELECT id AS visit_id, patient_id, recorded_at AS visited_at_after_appointment
+        FROM blood_sugars
+        WHERE deleted_at IS NULL
+          AND patient_id = patients_with_appointments.patient_id
+          AND patients_with_appointments.previous_appointment_date < blood_sugars.recorded_at
+          AND blood_sugars.recorded_at < patients_with_appointments.month_date + INTERVAL '1 month' + INTERVAL '15 days'
+        UNION
+        (
+            SELECT id AS visit_id, patient_id, recorded_at AS visited_at_after_appointment
+            FROM blood_pressures
+            WHERE deleted_at IS NULL
+              AND patient_id = patients_with_appointments.patient_id
+              AND patients_with_appointments.previous_appointment_date < blood_pressures.recorded_at
+              AND blood_pressures.recorded_at < patients_with_appointments.month_date + INTERVAL '1 month' + INTERVAL '15 days'
```

Comment:
`15 days` is the ideal interval for a patient to return back to care because of a call.  We currently consider the calls made to overdue patients until the end of the month. 
`+ 1 month + 15 days` This interval ensures we have given enough time for a patient who got a call at the end of the month to return to care.